kimchi/src/prover.rs

```diff
@@ -85,11 +85,26 @@ impl<G: CommitmentCurve> ProverProof<G>
 where
     G::BaseField: PrimeField,
 {
-    /// This function constructs prover's zk-proof from the witness & the ProverIndex against SRS instance
+    /// This function constructs prover's non-recursive zk-proof from the witness & the ProverIndex against SRS instance
     ///     witness: computation witness
     ///     index: ProverIndex
     ///     RETURN: prover's zk-proof
     pub fn create<EFqSponge: Clone + FqSponge<Fq<G>, G, Fr<G>>, EFrSponge: FrSponge<Fr<G>>>(
+        groupmap: &G::Map,
+        witness: [Vec<Fr<G>>; COLUMNS],
+        index: &ProverIndex<G>,
+    ) -> Result<Self> {
+        Self::create_recursive::<EFqSponge, EFrSponge>(groupmap, witness, index, Vec::new())
+    }
```

Comment:
I think you'll have to change the way `create_recursive` work, can you modify a test (for example `tests/generic.rs`) to use the non-recursive create? Let's see if this passes tests :o

Reply:
changed tests/generic.rs verify_proof function to use non-recursive create, (commented out the prev argument), test still passes which is weird, it's as tho prev not being present has no effect.

Reply:
looked at the code, looks like the recursion stuff doesn't do anything if the vector is empty! Can you add the test as a commit here, LGTM otherwise :)